### PR TITLE
remove the deprecated extra_credentials endpoints

### PR DESCRIPTION
**Pull request by ryanpetrello**
cc @gamuniz

### CODE REVIEW

File: awx/api/views/__init__.py [hunk 1]
```diff
@@ -2344,51 +2344,6 @@ def modernize_launch_payload(self, data, obj):
         if 'inventory' not in modern_data and id_fd in modern_data:
             modern_data['inventory'] = modern_data[id_fd]
 
-        # Automatically convert legacy launch credential arguments into a list of `.credentials`
```
Comment (ryanpetrello) [inline awx/api/views/__init__.py:2347]: 🔪🔪🔪🔪🔪

File: docs/prompting.md [hunk 2]
```diff
@@ -285,5 +285,4 @@ of what happened.
    - Spawned job still uses all job template credentials
 
 **Credentials Deprecated Behavior**
-   - Manual launch providing `"extra_credentials": []` should launch with no job credentials
```
Comment (AlanCoding) [inline docs/prompting.md:288]: remove whole section

**Comment by shanemcd** [comment]
recheck

**Comment by shanemcd** [comment]
recheck

File: awx/api/views/__init__.py [hunk 3]
```diff
@@ -2344,51 +2344,6 @@ def modernize_launch_payload(self, data, obj):
...
-            # combine the list of "new" and the filtered list of "old"
-            new_credentials.extend([cred.pk for cred in existing_credentials])
-            if new_credentials:
-                # If provided list doesn't contain the pre-existing credentials
-                # defined on the template, add them back here
-                for cred_obj in template_credentials:
-                    if cred_obj.pk not in new_credentials:
-                        new_credentials.append(cred_obj.pk)
-                modern_data['credentials'] = new_credentials
-
```
Comment (AlanCoding) [inline awx/api/views/__init__.py:2391]: `ignored_fields` returned as the 2nd item in the tuple from this method is always `{}`. But it looks like that was pre-existing...

**Review by AlanCoding — APPROVED** [review]

**Comment by wenottingham** [comment]
Idea of removing the deprecated endpoints is OK with me. Didn't check the code.

**Comment by ryanpetrello** [comment]
regate

**Pull request merged**